src/clib/pio_file.c

```diff
@@ -405,9 +417,9 @@ int PIOc_sync (int ncid)
   msg = PIO_MSG_SYNC;
 
   if(ios->async_interface && ! ios->ioproc){
-    if(ios->compmaster) 
+    if(!ios->comp_rank) 
```

Comment:
! ios->comp_rank is confusing - isn't ios->compmaster the boolean for this?  Otherwise use ios->comp_rank == 0.